.github/workflows/inspektor-gadget.yml

```diff
@@ -697,6 +697,48 @@ jobs:
           rm -rf /tmp/.buildx-cache
           mv /tmp/.buildx-cache-new /tmp/.buildx-cache
 
+  analyze-and-build:
+    name: IG binary size analysis
+    runs-on: ubuntu-latest
+    needs: 
+      - build-ig 
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@3041bf56c941b39c61721a86cd11f3bb1338122a # v5.2.0
+        with:
+          go-version: '1.23.4'
+          cache: true
+        id: go
+      - name: Install go-size-analyzer
+        run: |
+          go install github.com/Zxilly/go-size-analyzer/cmd/gsa@e47aa2561e70b937f2be7441b2eeca984ff60e8b
+      - name: Get ig-linux-amd64.tar.gz from artifact.
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: ig-linux-amd64-tar-gz
+          path: /home/runner/work/inspektor-gadget/
+      - name: Unpack ig-linux-amd64.tar.gz
+        run: |
+          tar zxvf /home/runner/work/inspektor-gadget/ig-linux-amd64.tar.gz
+      - name: Generate JSON
+        run: |
+          gsa ./ig -f "json" --compact | tee out.json
+      - name: Extract Packages as JSON Array
+        run: |
+          jq '[.packages | to_entries[] | {package: .key, size: .value.size, type: .value.type}] | sort_by(.size) | reverse' out.json | tee ig-packages.json
+        shell: bash
```

Comment:
IMO these three can be in a single step. Any reason to have these too granular steps?